feat: Add Reusable Button Component with Icon and Loader Support

Honey-Sharma-git opened the pull request:
_This PR strives to add reusable button component_
**<Button /> Component:**
A customizable and reusable button that accepts the following props:
- children: Button label
- className: For custom styling
- onClick: Event handler
- isLoading: Boolean to show loading state
- iconPath: Optional icon (e.g., Add or Export)

**<InlineLoader /> Component:**

- Displays a spinner inline within the button during loading state.

**SVG Icons:**

- Add icon
- CSV file (Export) icon

**Why This Change**
This component streamlines the implementation of consistent UI buttons across the app, reduces duplication, and provides better UX by handling loading and icon use cases internally.

Honey-Sharma-git committed:
feat: Add Button component
- Add icons for adding and CSV file.
- Add <InlineLoader/> component to show loader inside buttons.
- Add reusable <Button/> that takes children, className, onClick, isLoading, iconPath as props.

manager-honey commented on src/components/Button.tsx:
```diff
@@ -0,0 +1,37 @@
+  return (
+    <button
+      onClick={onClick}
+      className={`bg-blue-500 py-2 px-4  rounded-md min-w-18 cursor-pointer hover:brightness-90 focus:outline-gray-400 focus:shadow ${className}`}
```
Why is there a background color class hardcoded not customizable?

manager-honey replied:
Why is there a background color class hardcoded not customizable?

manager-honey commented on src/components/Button.tsx:
```diff
@@ -0,0 +1,37 @@
+  return (
+    <button
+      onClick={onClick}
+      className={`bg-blue-500 py-2 px-4  rounded-md min-w-18 cursor-pointer hover:brightness-90 focus:outline-gray-400 focus:shadow ${className}`}
```
Why is there a background color class hardcoded not customizable?

Honey-Sharma-git replied:
This issue will be fixed in the next PR with more support

manager-honey commented on src/components/Button.tsx:
```diff
@@ -0,0 +1,37 @@
+    >
+      <span className="flex flex-row items-center  justify-center gap-2">
+        {isLoading ? (
+          <InlineLoader />
```
InlineLoader uses div, a block level element. A block level element should not be nested in inline level element like span.

Honey-Sharma-git replied:
This issue has been fixed.

manager-honey commented on src/components/InlineLoader.tsx:
```diff
@@ -0,0 +1,5 @@
+export const InlineLoader = () => {
+  return (
+    <div className="aspect-square h-5 border-2 border-t-transparent animate-spin rounded-full border-currentcolor"></div>
```
Use inline level element span as it would align more with the name of the component.

Honey-Sharma-git replied:
This issue has been fixed.

manager-honey commented on public/icons/CSV.svg:
If we are using svgs, why not use them with a wrapper component to make them more customizable.

Honey-Sharma-git replied:
I have delete them since they were used for testing purpose and pushed accidently.

manager-honey commented on src/components/Button.tsx:
```diff
@@ -0,0 +1,37 @@
+  className?: string;
+  iconPath?: string;
+  onClick?: () => void;
+  isLoading?: boolean;
```
Use rest operator to accept other event handler through the props.

Honey-Sharma-git replied:
This issue has been fixed. I have removed the onClick explicit prop as it can be now used through ...rest prop.

Honey-Sharma-git committed:
fix: fix miscellaneous bugs
- Remove unused CSV.svg and add.png images.
- Make children prop mandatory in Button.tsx component.
- Extended button props to take all valid HTML button attributes and remove explicit onClick prop.
- Remove block level element's nesting under inline element.
- Change div to span in InlineLoader to match the name of the component.
